### PR TITLE
 Feature/dsp 393 support s3 csv to dataframe

### PR DESCRIPTION
**Pull request by zhangchi1**
**What this PR does / why we need it**:
1. Add support for reading csv and gzip csv files in s3 to dataframe for python client
**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: sdk/python/feast/job.py [hunk 1]
```diff
@@ -128,14 +132,51 @@ def result(self, timeout_sec: int = int(defaults[CONFIG_TIMEOUT_KEY])):
...
             Iterable of Avro rows.
         """
-        uris = self.get_avro_files(timeout_sec)
+        # list of parseResult
+        uris = self.get_files(timeout_sec)
         for file_uri in uris:
             file_obj = get_staging_client(file_uri.scheme).download_file(file_uri)
             file_obj.seek(0)
-            avro_reader = fastavro.reader(file_obj)
+            print("===============In result============")
```
Comment (swatiarora0208) [inline sdk/python/feast/job.py]: remove print statements after testing is done

**Review by swatiarora0208 — APPROVED** [review]

File: sdk/python/feast/job.py [hunk 2]
```diff
@@ -90,7 +94,7 @@ def get_avro_files(self, timeout_sec: int = int(defaults[CONFIG_TIMEOUT_KEY])):
                 is exceeded, an exception will be raised.
 
         Returns:
-            str: Google Cloud Storage file uris of the returned Avro files.
+            str: Google Cloud Storage file uris of the returned Avro files or CSV files.
```
Comment (jmelinav) [inline sdk/python/feast/job.py]: could be s3 or gcs right?

Comment (zhangchi1) [reply sdk/python/feast/job.py]: correct, let me update this comment

File: sdk/python/tests/test_client.py [hunk 3]
```diff
@@ -794,20 +794,20 @@ def test_apply_feature_set_success(self, test_client):
 
         # List Feature Sets
         assert (
```
Comment (jmelinav) [inline sdk/python/tests/test_client.py]: undo the formatting.

File: sdk/python/feast/job.py [hunk 4]
```diff
@@ -128,14 +132,48 @@ def result(self, timeout_sec: int = int(defaults[CONFIG_TIMEOUT_KEY])):
...
+                with gzip.open(file_obj, mode='rt') as f:
+                    csv_reader = csv.DictReader(f, delimiter=',')
+                    for record in csv_reader:
+                        yield record
+            else:
+                raise Exception(
+                    "Feast only supports CSV data format with .csv or .csv.gz extension for now. Please check "
+                    "your Feast Serving deployment."
+                )
+        else:
```
Comment (jmelinav) [inline sdk/python/feast/job.py]: is this necessary?

File: sdk/python/tests/test_client.py (hunk 3 above)
Comment (jmelinav) [reply sdk/python/tests/test_client.py]: same in other places where there is just reformatting and no changes.

File: sdk/python/feast/job.py (hunk 4 above)
Comment (zhangchi1) [reply sdk/python/feast/job.py]: I think we need to add an exception case to check when the file is not in any of the above file formats

**Pull request merged**